sql/src/main/java/io/crate/execution/engine/pipeline/ProjectionToProjectorVisitor.java

```diff
@@ -233,8 +237,19 @@ public Projector visitOrderedTopN(OrderedTopNProjection projection, Context cont
         for (int i = numOutputs; i < inputs.size(); i++) {
             orderByIndices[idx++] = i;
         }
+
+        int rowMemoryOverhead = 32; // priority queues implementation are backed by an arrayList
+        RowCellsAccountingWithEstimators rowAccounting = new RowCellsAccountingWithEstimators(
+            Symbols.typeView(Stream.concat(projection.outputs().stream(),
```

Comment:
```suggestion
            Symbols.typeView(Lists2.concat(projection.outputs(), projection.orderBy()));
```

Would avoid the stream creations and size the list correctly (opposed to the `.toList()` collector)

Reply:
YES! browsed the Collections/List for something like this